colcon_powershell/shell/template/prefix.ps1.em

```diff
@@ -3,19 +3,23 @@
 # This script extends the environment with all packages contained in this
 # prefix path.
 
-# use the Python executable known at configure time
-$_colcon_python_executable="@(python_executable)"
-# allow overriding it with a custom location
+# check environment variable for custom Python executable
 if ($env:COLCON_PYTHON_EXECUTABLE) {
-  $_colcon_python_executable="$env:COLCON_CURRENT_PREFIX"
-}
-# if the Python executable doesn't exist try another fall back
-if (!(Test-Path "$_colcon_python_executable" -PathType Leaf)) {
-  if (Get-Command "python" -ErrorAction SilentlyContinue) {
+  if (!(Test-Path "$env:COLCON_PYTHON_EXECUTABLE" -PathType Leaf)) {
+    echo "error: COLCON_PYTHON_EXECUTABLE '$env:COLCON_PYTHON_EXECUTABLE' doesn't exist"
+    exit 1
+  }
+  $_colcon_python_executable="$env:COLCON_PYTHON_EXECUTABLE"
+} else {
+  # use the Python executable known at configure time
+  $_colcon_python_executable="@(python_executable)"
+  # if it doesn't exist try a fall back
+  if (!(Test-Path "$_colcon_python_executable" -PathType Leaf)) {
+    if (!(Get-Command "python" -ErrorAction SilentlyContinue)) {
```

Comment:
`python` is actually not a generic enough fall back. On Linux that will try a Python 2 interpreter (which will fail to run the Python code)...

Reply:
See ed8f05b61768270a7d4a6158aceda2587d3c2e59